pkg/cmd/cli/cmd/observe/observe.go

```diff
@@ -0,0 +1,957 @@
+
+	// the commands to execute on changes
+	cmd.Flags().VarP(&options.eachCommand, "each", "e", "A command to run for each change. Specify mulitple times to add arguments.")
+
+	// to perform deletion synchronization
+	cmd.Flags().VarP(&options.deleteCommand, "delete", "d", "A command to run when resources are deleted. Specify mulitple times to add arguments.")
+	cmd.Flags().Var(&options.nameSyncCommand, "names", "A command that will list all of the currently known names, optional. Specify multiple times to add arguments. Use to get notifications when objects are deleted.")
+
+	// add additional arguments / info to the server
+	cmd.Flags().StringVar(&options.templateType, "use", options.templateType, "Format for template arguments. jsonpath and gotemplate are supported.")
+	cmd.Flags().BoolVar(&options.strictTemplates, "strict-templates", false, "If true return an error on any field or map key that is not missing in a template.")
+	cmd.Flags().VarP(&options.templates, "argument", "a", "Template for the arguments to be passed to each command in the format defined by --use.")
+	cmd.Flags().StringVar(&options.typeEnvVar, "type-env-var", "", "The name of an env var to set with the type of event received ('Sync', 'Updated', 'Deleted', 'Added') as the first argument to the --each command.")
+	cmd.Flags().StringVar(&options.objectEnvVar, "object-env-var", "", "The name of an env var to serialize the object to when calling --each, optional.")
+
+	// control retries of individual commands
+	cmd.Flags().IntVar(&options.maximumErrors, "maximum-errors", options.maximumErrors, "Exit after this many errors have been detected with. May be set to -1 for no maximum.")
+	cmd.Flags().IntVar(&options.retryExitStatus, "retry-on-exit-status", 0, "If set when the each, names, or sync commands returns this exit code, retry up to --retry-count times.")
+	cmd.Flags().IntVar(&options.retryCount, "retry-count", options.retryCount, "The number of times to retry a failing command before continuing.")
+
+	// control observe program behavior
+	cmd.Flags().DurationVar(&options.exitAfterPeriod, "exit-after", 0, "Exit with status code 0 after the provided duration, optional.")
+	cmd.Flags().DurationVar(&options.resyncPeriod, "resync-period", 0, "When non-zero, periodically reprocess every item from the server as a Sync event. Use to ensure external systems are kept up to date. Requires --names")
+	cmd.Flags().BoolVar(&options.printMetricsOnExit, "print-metrics-on-exit", false, "On exit write all metrics to stdout.")
+	cmd.Flags().StringVar(&options.listenAddr, "listen-addr", options.listenAddr, "The name of an interface to listen on to expose metrics and health checking.")
+
+	// additional debug output
+	cmd.Flags().BoolVar(&options.noHeaders, "no-headers", false, "Skip printing information about each event prior to executing the command.")
```

Comment:
> I think in real usage you actually do want to get a log of all
> notifications.  I also don't think glog format is what most people want for
> this.  And you want a delineation between each command invocation.
>  --no-headers is when your script owns writing out descriptive info, or you
> don't want any output.

I think you'll lose the ordering argument as soon as you allow parallel execution, which seems like something we're likely to want.